### ISSUE TITLE
beIdenticalTo and Swift structs

### ISSUE THREAD

**Issue by lukeredpath** [Quick/Nimble#298]
Hi, new user of Nimble here. I might be missing something obvious here but I don't seem to be able to use `beIdenticalTo` how I'd expect. Does it work with Swift types?

```swift
let identity = ActorIdentity(PINCode: "1234", identityToken: "ABCD")
expect(identity).to(beIdenticalTo(identity))
```

The above does not compile. ActorIdentity is a struct. The compiler error is `Cannot invoke beIdenticalTo with an argument list of type (ActorIdentity)`.

If this isn't possible then how can I write an expectation that checks that the struct instance returned is what I expected?

**Comment by ashfurrow** [comment]
Switch structs are value types, and (at first glance) not compatible with `beIdenticalTo` _because_ they're value types. Consider:

```swift
var a = ActorIdentity(PINCode: "1234", identityToken: "ABCD")
var b = a

// a and b are now two different values, even if they hold the same struct fields.

a === b // compiler error, because they're two different types.

b.PINCode = "0000"

// It's now more clear that a and b are not identical, but they never were.
```

If you're looking for equality comparison, you can create an `==` operator for your struct and use `expect(identity).to( equal(identity) )`. Does that answer your question?

**Comment by lukeredpath** [comment]
Hi Ash, yes, that makes sense now. I ended up checking the identityToken on each one as it felt a bit wrong to define equality just for my tests (I had no use case for it elsewhere). Thanks.

Its possible that this was just me being a bit slow today but I wonder if it would be helpful to clarify this in the documentation.

**Comment by ashfurrow** [comment]
I agree, documentation could definitely be improved. There used to be a note in the Readme about this matcher only working on Objective-C-compatible objects, but was removed with Swift 2. I think adding back in a comment about the matcher only working with _classes_ would be great. Would you be able to send a PR? The Readme would be the place to put it, maybe elsewhere if you think it would be helpful?

**Comment by lukeredpath** [comment]
@ashfurrow sure, first pass above.

**Comment by ashfurrow** [comment]
Fixed in #299.
